pandas/tests/io/test_clipboard.py

```diff
@@ -243,6 +244,34 @@ def test_read_clipboard_infer_excel(self, request, mock_clipboard):
 
         tm.assert_frame_equal(res, exp)
 
+    def test_infer_excel_with_nulls(self, request, mock_clipboard):
+        # GH41108
+        text = "col1\tcol2\n1\tred\n\tblue\n2\tgreen"
+
+        mock_clipboard[request.node.name] = text
+        df = read_clipboard()
+        df_expected = DataFrame(
+            data={"col1": [1, None, 2], "col2": ["red", "blue", "green"]}
+        )
+
+        # excel data is parsed correctly
+        tm.assert_frame_equal(df, df_expected)
+
+    def test_infer_excel_with_multiindex(self, request, mock_clipboard):
+        # GH41108
+        text = "\t\tcol1\tcol2\nA\t0\t1\tred\nA\t1\t\tblue\nB\t0\t2\tgreen"
```

Comment:
It can be that the second row does not have A as well. Removing A here, I think the test fails, but it also fails on master. Okay to raise an issue about clipboard supporting MultiIndex better (if one doesn't already exist)

Reply:
It doesn't seem to be failing when I try it with this:

```
\t\tcol1\tcol2
A\t0\t1\tred
\t1\t\tblue
B\t0\t2\tgreen
```

Is this what you mean? maybe i'm not understanding it correctly